pkg/virt-launcher/virtwrap/network/podinterface.go

```diff
@@ -1103,11 +1103,42 @@ func (b *MasqueradeBindMechanism) createNatRulesUsingNftables(proto iptables.Pro
 		return err
 	}
 
+	addressesToDnat, err := b.getDstAddressesToDnat(proto)
+	if err != nil {
+		return err
+	}
+
 	if len(b.iface.Ports) == 0 {
-		err = b.handler.NftablesAppendRule(proto, "nat", "KUBEVIRT_PREINBOUND",
+		if !hasIstioSidecarInjectionEnabled(b.vmi) {
+			err = b.handler.NftablesAppendRule(proto, "nat", "KUBEVIRT_PREINBOUND",
+				"counter", "dnat", "to", b.getDhcpConfigIpByProtocol(proto))
+			if err != nil {
+				return err
+			}
+		}
+
+		if hasIstioSidecarInjectionEnabled(b.vmi) {
+			err = b.skipForwardingForReservedPortsUsingNftables(proto, portsUsedByIstio())
+			if err != nil {
+				return err
+			}
+		}
+
+		err = b.handler.NftablesAppendRule(proto, "nat", "KUBEVIRT_POSTINBOUND",
+			b.handler.GetNFTIPString(proto), "saddr", b.getSrcAddressesToSnat(proto),
+			"counter", "snat", "to", b.getGatewayByProtocol(proto))
+		if err != nil {
+			return err
+		}
+
+		err = b.handler.NftablesAppendRule(proto, "nat", "output",
+			b.handler.GetNFTIPString(proto), "daddr", addressesToDnat,
 			"counter", "dnat", "to", b.getDhcpConfigIpByProtocol(proto))
+		if err != nil {
+			return err
+		}
```

Comment:
This has to be done only in the case of istio proxy right ?

Reply:
Well, it is now. These rules are coincidentally the same for `kubectl port-forward` and for Istio. There is a effort in this PR by Alon https://github.com/kubevirt/kubevirt/pull/5431 to make `kubectl port-forward` work for VMs without explicit ports.
His PR adds these rules, similarly as they are added in the VM with explicit ports.

I think I have two options:
1) I could `return nil` as you suggested [here](https://github.com/kubevirt/kubevirt/pull/5461#discussion_r634219474), because this PR doesn't address `kubectl port-forward`.
In this case, Alon would need to remove that `return` in his PR.
2) wait for Alon's PR to get in first

Reply:
> Well, it is now. These rules are coincidentally the same for `kubectl port-forward` and for Istio. There is a effort in this PR by Alon #5431 to make `kubectl port-forward` work for VMs without explicit ports.
> His PR adds these rules, similarly as they are added in the VM with explicit ports.
> 
> I think I have two options:
> 
> 1. I could `return nil` as you suggested [here](https://github.com/kubevirt/kubevirt/pull/5461#discussion_r634219474), because this PR doesn't address `kubectl port-forward`.
>    In this case, Alon would need to remove that `return` in his PR.
> 2. wait for Alon's PR to get in first

I would go with 1 so we reduce the scope of this PR, whoever is merged first will conflict and will need to be rebased.

Reply:
[Done](https://github.com/kubevirt/kubevirt/compare/2ae68e9c5d93b540a80e5c6b931418c2f96edd97..0c67e2ea22d9a6a185bc2c84f1b6b295a581159c)

Reply:
What do you think about moving the Istio aware code on top and the non istio code after it ? since usually istio is not deployed it feels better to have it under an if statement